zerver/lib/message.py

```diff
@@ -586,7 +586,8 @@ def has_message_access(
 
 
 def event_recipient_ids_for_action_on_messages(
-    messages: list[Message],
+    message_ids: list[int],
+    is_channel_message: bool,
```

Comment:
Probably worth a docstring/comment on why this is required, and other updates -- it's not immediately obvious from the signature why this is a parameter, and not just imputed from the `channel` parameter, for example. 

(I think the actual architecture is good).